helmexec/runner.go

```diff
@@ -23,5 +24,7 @@ type ShellRunner struct {
 func (shell ShellRunner) Execute(cmd string, args []string) ([]byte, error) {
 	preparedCmd := exec.Command(cmd, args...)
 	preparedCmd.Dir = shell.Dir
+	preparedCmd.Env = os.Environ()
+	preparedCmd.Env = append(preparedCmd.Env, "HELM_TILLER_SILENT=true")
 	return preparedCmd.CombinedOutput()
```

Comment:
Very nit but you should also alter `KUBECONFIG` to an absolute path if it is already set, otherwise `helm-tiller` is unable to connect the cluster.

```
# This doesn't work AFAIK
KUBECONFIG=kubeconfig helm tiller run -- helm list

# This works
KUBECONFIG=$(pwd)/kubeconfig helm tiller run -- helm list
```

Perhaps something like this is needed:

```
kubeconfig := os.Getenv("KUBECONFIG")
if kubeconfig != "" {
    abs, err := filepath.Abs(kubeconfig)
    // err handling...
    if !abs {
        wd, err := os.Getwd()
        if err != nil {
            return ....
        }
        preparedCmd.Env = append(preparedCmd.Env, "KUBECONFIG=" + filepath.Join(wd, kubeconfig))
    }
    // ....
}
```

Reply:
OK, I ended up with a very different code, but it's done.